Add docs for MKE and eBPF (#1556)
tigera/docs: calico-enterprise/operations/ebpf/enabling-ebpf.mdx

- [Verify that your cluster is ready for eBPF mode](#verify-that-your-cluster-is-ready-for-ebpf-mode)
- [Configure {{prodname}} to talk directly to the API server](#configure-calico-enterprise-to-talk-directly-to-the-api-server)
- [Configure kube-proxy](#configure-kube-proxy)

> - [(MKE only) Disable host conntrack bypass](#disable-host-conntrack-bypass)

Reviewer: Do we have a rule that the link and the heading should be the same?

Author: I have removed this section as it is adding to the confusion. operator already sets ```bpfHostConntrackBypass``` to ```false```.

> - [Disable host conntrack bypass in MKE](#disable-host-conntrack-bypass-in-mke)

Reviewer: ```suggestion
- [(MKE only) Disable host conntrack bypass](#disable-host-conntrack-bypass-in-mke)
```
I find it best to put the conditional part of the sentence up front so the user can skip ahead if this doesn't apply them.  Otherwise they're likely to read the first half of the sentence and start doing it before they reach the "but only if you're on MKE" part

  `api-int.<cluster_name>.<base_domain>` should point to the API server or to the load balancer in front of the
  API server. Use that (filling in the `<cluster_name>` and `<base_domain>` as appropriate for your cluster) for the
  `KUBERNETES_SERVICE_HOST` below. Openshift uses 6443 for the `KUBERNETES_SERVICE_PORT`.

> - MKE runs a reverse proxy in each node, that can be used to reach the API server. You should use `proxy.local`

Reviewer 2: ```suggestion
- MKE runs a reverse proxy in each node that can be used to reach the API server. You should use `proxy.local`
```

> - MKE runs a reverse proxy in each node, which can be used to reach the API server. You should use `proxy.local`

Reviewer: Wait, didn't we have problems with pod-networked pods when using that?  Has that been fixed in operator?

Author: After setting ```bpfHostConntrackBypass``` to ```false```, I did not see any issue with pod networked pods coming up.

> - MKE runs a reverse proxy in each node, which can be used to reach the API server. You should use `proxy.local`

Reviewer 2: ```suggestion
- MKE runs a reverse proxy in each node that can be used to reach the API server. You should use `proxy.local`
```

kubectl patch networks.operator.openshift.io cluster --type merge -p '{"spec":{"deployKubeProxy": true}}'
```

#### MKE

If you are running MKE, you can disable `kube-proxy` as follows:

> Download the cluster configuration, set `kube_proxy_mode` to `disabled` and `kube_default_drop_masq_bits` to `true` and upload the cluster configuration.

Reviewer: Setting `kube_default_drop_masq_bits` is mandatory, I think.  BPF is incompatible without that whether you disable kube-proxy or not.

> Download the cluster configuration, set `kube_proxy_mode` to `disabled` and `kube_default_drop_masq_bits` to `true` and upload the cluster configuration.

Reviewer: "Download the cluster configuration" is a bit vague, can you either make this a link to the MKE docs or give the command to do it?

> Download the cluster configuration, set `kube_proxy_mode` to `disabled` and `kube_default_drop_masq_bits` to `true` and upload the cluster configuration.

Reviewer 2: I think something along these lines would work a bit more smoothly:

Follow the step procedure in [Modify an existing MKE configuration](link) to download, edit, and upload your MKE configuration. During the editing step, add the following configuration:
```
config config
```

> Follow the step procedure in [Modify an existing MKE configuration](https://docs.mirantis.com/mke/3.7/ops/administer-cluster/configure-an-mke-cluster/use-an-mke-configuration-file.html#modify-an-existing-mke-configuration) to download, edit, and upload your MKE configuration. During the editing step, add the following configuration:

Reviewer 2: ```suggestion
Follow the step procedure in [Modify an existing MKE configuration](https://docs.mirantis.com/mke/current/ops/administer-cluster/configure-an-mke-cluster/use-an-mke-configuration-file.html#modify-an-existing-mke-configuration) to download, edit, and upload your MKE configuration. During the editing step, add the following configuration:
```

> Steps to update the cluster configuration can be found in https://docs.mirantis.com/mke/3.7/ops/administer-cluster/configure-an-mke-cluster/use-an-mke-configuration-file.html#modify-an-existing-mke-configuration.

Reviewer 2: ```suggestion
Steps to update the cluster configuration can be found in [Modify and existing MKE configuration](https://docs.mirantis.com/mke/current/ops/administer-cluster/configure-an-mke-cluster/use-an-mke-configuration-file.html#modify-an-existing-mke-configuration).
```
A better link, for longevity.

If both `kube-proxy` and `BPFKubeProxyIptablesCleanupEnabled` is enabled then `kube-proxy` will write its iptables rules and Felix will try to clean them up resulting in iptables flapping between the two.

### Disable host conntrack bypass

> If you running MKE, host conntrack should not be bypassed. This can be done with `kubectl` as follows:

Reviewer 2: ```suggestion
If you running are MKE, host conntrack should not be bypassed. This can be done with `kubectl` as follows:
```

> If you running MKE, host conntrack should not be bypassed. This can be done with `kubectl` as follows:

Reviewer: Can we make operator set this flag in MKE mode?  Better to write code than docs!

Author: operator sets ```bpfHostConntrackBypass``` to ```false``` only if the user hasn't explicitly set this config to ```true```. I am fine with removing this step from the docs.

Reviewer 2: So we're removing everything from this section? ### Diasable host conntrack bypass in MKE

Author: @fasaxc Can we remove this step?

Reviewer 2: If we need to retain this, it might function better as a prerequisite.

   ```bash
   kubectl patch ds -n kube-system kube-proxy --type merge -p '{"spec":{"template":{"spec":{"nodeSelector":{"non-calico": null}}}}}'
   ```

> 2. If you are running MKE, download the cluster config as mentioned above, set `kube_proxy_mode` to `iptables` and `kube_default_drop_masq_bits ` to `false`

Reviewer 2: Do we really need people to download, edit, and upload this configuration twice? Is there not a way to combine it in one step?

Author: This is under reversing the process, where we move from bpf to iptables.

   ```bash
   kubectl patch ds -n kube-system kube-proxy --type merge -p '{"spec":{"template":{"spec":{"nodeSelector":{"non-calico": null}}}}}'
   ```
2. If you are running MKE, follow the step procedure in [Modify an existing MKE configuration](https://docs.mirantis.com/mke/current/ops/administer-cluster/configure-an-mke-cluster/use-an-mke-configuration-file.html#modify-an-existing-mke-configuration) to download, edit, and upload your MKE configuration. During the editing step, add the following configuration:

> `kube_proxy_mode` to `iptables` and `kube_default_drop_masq_bits ` to `false`.

Reviewer: > `kube_default_drop_masq_bits ` to `false`.

Are we sure that's the right default even for iptables mode?  Calico's default is to use 0xffff0000 bits in both modes but in BPF mode it cannot be changed.  So, if we have a conflict in BPF mode, we might have a conflict in iptables mode too?

Author: I checked with Vikram from Mirantis and he said its ok to set ```kube_default_drop_masq_bits``` to ```true``` even in iptables mode.

Author: Will make the changes in the doc accordingly.
